server/src/main/java/io/druid/server/coordinator/helper/DruidCoordinatorSegmentCompactor.java

```diff
@@ -150,8 +150,11 @@ private CoordinatorStats doRun(
 
       if (segmentsToCompact.size() > 1) {
         final DataSourceCompactionConfig config = compactionConfigs.get(dataSourceName);
+        // Currently set keepSegmentGranularity to false because it breaks the algorithm of CompactionSegmentIterator to
```

Comment:
Do you plan to address this in a future patch?

Reply:
Yes. I'll do in a follow-up PR.

Reply:
Raised https://github.com/apache/incubator-druid/issues/6135.